### PR TITLE
fix(sfs/file_system): remove multiple delete logic

### LINKED ISSUE #2189: [SFS] The action fobidden error of delete operation
The delete request is sent twice after `terraform destroy`, the second time will occur error.
```
Error: Error deleting Huaweicloud Share File: Action forbidden: [DELETE https://sfs.cn-north-4.myhuaweicloud.com/v2/0970dd7a1300f5672ff2c003c60ae115/shares/545b0ee5-53cd-4e06-a753-e2eb317b9687], error message: {"forbidden": {"message": "Invalid share: Share status must be one of (available,error,inactive).", "code": 403}}
```

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
When we delete an existing file system, the DELETE request will sent twice, and the second time will occur an error.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #2189 

**Special notes for your reviewer**:

**Release note**:

```release-note
1. refactor state refresh function.
```

## PR Checklist

* [x] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/fgs' TESTARGS='-run=TestAccFgsV2Function_agency'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/fgs -v -run=TestAccFgsV2Function_agency -timeout 360m -parallel 4
=== RUN   TestAccFgsV2Function_agency
=== PAUSE TestAccFgsV2Function_agency
=== CONT  TestAccFgsV2Function_agency
--- PASS: TestAccFgsV2Function_agency (232.07s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/fgs       232.142s
```

### CODE REVIEW

File: huaweicloud/resource_huaweicloud_sfs_file_system_v2.go [hunk 1]
```diff
@@ -409,67 +409,42 @@ func resourceSFSFileSystemV2Delete(d *schema.ResourceData, meta interface{}) err
 		return fmtp.Errorf("Error creating Huaweicloud Shared File Client: %s", err)
 	}
 
+	err = shares.Delete(sfsClient, d.Id()).ExtractErr()
+	if err != nil {
+		return fmtp.Errorf("Error deleting Huaweicloud Share File: %s", err)
```
Comment (Jason-Zhang9309) [inline huaweicloud/resource_huaweicloud_sfs_file_system_v2.go:414]: use lower case 'error' and remove 'Huaweicloud'

Comment (Lance52259) [reply huaweicloud/resource_huaweicloud_sfs_file_system_v2.go:414]: Keep style with other error return same and these codes will refactor in the next redesign PR.

File: huaweicloud/resource_huaweicloud_sfs_file_system_v2.go [hunk 2]
```diff
@@ -409,67 +409,42 @@ func resourceSFSFileSystemV2Delete(d *schema.ResourceData, meta interface{}) err
...
 			if _, ok := err.(golangsdk.ErrDefault404); ok {
-				logp.Printf("[INFO] Successfully deleted Huaweicloud shared File %s", shareId)
-				return r, "deleted", nil
+				return n, "deleted", nil
 			}
-			if errCode, ok := err.(golangsdk.ErrUnexpectedResponseCode); ok {
-				if errCode.Actual == 409 {
-					return r, "available", nil
-				}
+			if errCode, ok := err.(golangsdk.ErrUnexpectedResponseCode); ok && errCode.Actual == 409 {
```
Comment (niuzhenguo) [inline huaweicloud/resource_huaweicloud_sfs_file_system_v2.go]: This seems not needed anymore, it's used to check deleting error.

Comment (Lance52259) [reply huaweicloud/resource_huaweicloud_sfs_file_system_v2.go]: done.

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud' TESTARGS='-run=TestAccSFSFileSystemV2_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud -v -run=TestAccSFSFileSystemV2_ -timeout 360m -parallel 4
=== RUN   TestAccSFSFileSystemV2_basic
=== PAUSE TestAccSFSFileSystemV2_basic
=== RUN   TestAccSFSFileSystemV2_withEpsId
=== PAUSE TestAccSFSFileSystemV2_withEpsId
=== RUN   TestAccSFSFileSystemV2_withoutRule
=== PAUSE TestAccSFSFileSystemV2_withoutRule
=== CONT  TestAccSFSFileSystemV2_basic
=== CONT  TestAccSFSFileSystemV2_withoutRule
=== CONT  TestAccSFSFileSystemV2_withEpsId
--- PASS: TestAccSFSFileSystemV2_withoutRule (42.46s)
--- PASS: TestAccSFSFileSystemV2_withEpsId (61.12s)
--- PASS: TestAccSFSFileSystemV2_basic (117.21s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       117.317s
```

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/fgs' TESTARGS='-run=TestAccFgsV2Function_agency'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/fgs -v -run=TestAccFgsV2Function_agency -timeout 360m -parallel 4
=== RUN   TestAccFgsV2Function_agency
=== PAUSE TestAccFgsV2Function_agency
=== CONT  TestAccFgsV2Function_agency
--- PASS: TestAccFgsV2Function_agency (318.53s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/fgs       318.605s
```

File: huaweicloud/resource_huaweicloud_sfs_file_system_v2.go (hunk 2 above)
Comment (Jason-Zhang9309) [reply huaweicloud/resource_huaweicloud_sfs_file_system_v2.go]: If we remove this part, the retry of deleting will be removed. Please reconsider it.

**Pull request merged**